Sources/NIOPosix/NonBlockingFileIO.swift

```diff
@@ -627,4 +627,84 @@ public struct NonBlockingFileIO: NIOSendable {
         }
     }
 
+    public func lstat(path: String, eventLoop: EventLoop) -> EventLoopFuture<stat> {
+        return self.threadPool.runIfActive(eventLoop: eventLoop) {
+            let pointer = UnsafeMutablePointer<stat>.allocate(capacity: 1)
```

Comment:
No need for the stack allocated pointer, you can just do `var s = stat()` and then pass it in using `&`.

Reply:
🙏